Fixed package removal and removed unnecessary package.

mathuin opened the pull request:
The package xf86-video-odroid-c1 contributes a file to xorg.conf.d
which causes xinit to not work correctly in tests.  Removing this
package required an additional change to the installer to force
removals without asking for confirmation.

mathuin committed:
Fixed package removal and removed unnecessary package.

The package xf86-video-odroid-c1 contributes a file to xorg.conf.d
which causes xinit to not work correctly in tests.  Removing this
package required an additional change to the installer to force
removals without asking for confirmation.

mathuin commented:
@drinkcat this one passes the xorg arch test on the first try. :-)

drinkcat commented on targets/xorg:
```diff
@@ -104,10 +104,9 @@ END
-fi
+install xorg arch=,xserver-xorg-video-fbdev$ltspackages arch=,xserver-xorg-video-intel$ltspackages
+# JMT: eventually wrap this removal with an appropriate test
+remove xf86-video-odroid-c1
```
That's going to break non-ARM Chromebooks...

Instead of uninstalling stuff, I wonder if we could be a bit smarter about package selection, maybe install only `xorg-server`, `xorg-server-utils` and `xorg-apps`, and required drivers instead of the whole `xorg` group?

mathuin replied:
Ah, I misread that `if` -- I thought it was checking for `arch`-ness, not `arm`-ness!

I like the idea of installing only the required packages.  As new devices emerge (or older ones resurge), their drivers can be added as necessary.

mathuin committed:
Fixed bug introduced for non-ARM Arch systems.

I inadvertently removed the installation of intel video packages for
non-ARM Arch systems.  Fixed that!

mathuin commented:
I have fixed the inadvertent breakage here.

drinkcat commented on targets/xorg:
```diff
@@ -108,6 +108,8 @@ install xorg arch=,xserver-xorg-video-fbdev$ltspackages
     install arch=,xserver-xorg-video-intel$ltspackages
 fi
+# JMT: eventually wrap this removal with an appropriate test
+remove xf86-video-odroid-c1
```
That will still break on non-arm, non-arch: xf86-video-odroid-c1 does not exist on non-arm arch.

So you should put it in the else branch of the if statement above, and you probably need to change the line to `remove arch=xf86-video-odroid-c1,`. But, again, I'd rather change the install line and only install necessary drivers.

mathuin replied:
To install only necessary packages, we need two things:  one, a list of universally required packages; and two, a way to identify individual devices or features.  Your suggestion for the first (`xorg-server`, `xorg-server-utils` and `xorg-apps`) looks like a great start.  I have no idea how to do the second.  Is there an elegant way to check whether a device is an odroid, for instance?

drinkcat replied:
It's actually simpler than it looks (there aren't that many different graphics drivers needed on Chromebooks). You can just mirror what is done on Ubuntu/Debian: fbdev everywhere, intel on non-ARM.

`targets/xiwi` will also need fixing.
